openedx/core/djangoapps/user_api/accounts/image_helpers.py

```diff
@@ -102,19 +109,31 @@ def _get_default_profile_image_urls():
     return _get_profile_image_urls(
         settings.PROFILE_IMAGE_DEFAULT_FILENAME,
         staticfiles_storage,
-        file_extension=settings.PROFILE_IMAGE_DEFAULT_FILE_EXTENSION
+        file_extension=settings.PROFILE_IMAGE_DEFAULT_FILE_EXTENSION,
     )
 
 
-def set_has_profile_image(username, has_profile_image=True):
+def get_profile_image_version(username):
+    """
+    System (not user-facing) API call used to check the current version of
+    the user-uploaded profile image (if any).  Used by profile_image API.
+    """
+    try:
+        return UserProfile.objects.get(user__username=username).profile_image_version
+    except ObjectDoesNotExist:
+        raise UserNotFound()
+
+
+def set_profile_image_version(username, version):
     """
     System (not user-facing) API call used to store whether the user has
     uploaded a profile image.  Used by profile_image API.
     """
+    assert isinstance(version, (str, unicode, type(None)))
```

Comment:
Question: I thought we were encouraged to use pycontracts rather than `assert isinstance`.